app/models/annotations/tag.rb

```diff
@@ -89,7 +89,7 @@ def self.run_bulk_create_callbacks(ids_json, pmids_json)
   def get_tag_text_reference
     if self.tag.is_a?(String)
       team_id = self.team&.id
-      tag_text = TagText.where(text: self.tag, team_id: team_id).last
+      tag_text = TagText.where(text: self.tag.strip, team_id: team_id).last
```

Comment:
Also I suggest to trim database value so we can catch existing one with trailing space
i.e
tag_text = TagText.where(team_id: team_id).where('trim(text) = ?', self.tag.strip).last

Reply:
Oh I see that we normalize tag before saving in DB and did `self.text.strip.gsub(/^#/, '')` so we can skip this comment